SDK-2271. Added info about the Evaluator tool (#1806)
spryker/spryker-docs: docs/scos/dev/keeping-a-project-upgradable/upgradability-guidelines/dead-code.md

---
title: Dead code
description: Reference information for evaluator tools.
template: howto-guide-template
---

The checker that checks dead code that extends core classes in the project.

## Problem description

The project can extend core classes and after some time it can become unusable due to the core changes. As a result, it can cause issues related to the project's upgradability.

> It checks possible dead classes. For obvious reasons, the classes that are created magically tend to be skipped.

Reviewer: That is not the best way to describe it. 
It is about classes that are part of the spryker infrastructure and are created automatically by the spryker kernel. For the end user, it happens "magically" and they should not be worried about their creation. It looks like we need to rephrase it. 

```suggestion
It checks possible dead classes. For obvious reasons, the Spryker kernel classes such as `Factory`, `Facade` or `DependencyProvider` tend to be skipped.
```
